### PR TITLE
you can now use the mortar during hijack if you are not a coward + custom shell warning for ghosts

### PR DESCRIPTION
**Pull request by iloveloopers**
# About the pull request
lets you deploy the mortar shipside.
lets you fire a shell on a shipside mortar during hijack but do not expect to live.

ghosts now get a warning when an OT shell is about to land. (I did not test this and I don't know how to, I just copy pasted from CAS fire missions)

# Explain why it's good for the game
its funny and cool and soul

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

https://github.com/cmss13-devs/cmss13/assets/140007537/c7f7dcb2-98d1-438c-ba00-6a57f5ad9463

https://github.com/cmss13-devs/cmss13/assets/140007537/8862d7e1-cda9-4fda-9c2b-7b5c105fe3e0

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: You can now use the mortar while shipside during hijack, but only once.
add: Ghosts now get a warning when a custom mortar shell is about to land.
/:cl:

### CODE REVIEW

**Comment by HaultyAnonie** [comment]
So we're just gonna ignore the restrictions of space.

**Comment by joaopslima** [comment]
> So we're just gonna ignore the restrictions of space.

just aim the mortar horizontally

**Comment by Huffie56** [comment]
considering mortar and cas work on fiorina station i don't see why not mortar and cas on almayer

**Comment by LeDoktor** [comment]
Not a contributor, but this is the kind of shit CM NEEDS if it's going to survive another month. SPEEDMERGE THIS RIGHT NOW!!!!

**Comment by iloveloopers** [comment]
> considering mortar and cas work on fiorina station i don't see why not mortar and cas on almayer. not sure why you limiting yourself to one shot and then it kill you?

Fiorina has glass ceilings, the shell wouldn't go through the almayer's metal ceiling

File: code/modules/cm_marines/equipment/mortar/mortars.dm [hunk 1]
```diff
@@ -209,37 +211,44 @@
 /obj/structure/mortar/attackby(obj/item/O, mob/user)
 	if(istype(O, /obj/item/mortar_shell))
 		var/obj/item/mortar_shell/mortar_shell = O
+		var/turf/T = locate(targ_x + dial_x + offset_x, targ_y + dial_y + offset_y, z)
+		var/area/A = get_area(T)
```
Comment (Zonespace27) [inline code/modules/cm_marines/equipment/mortar/mortars.dm:215]: no one letter vars

File: code/modules/cm_marines/equipment/mortar/mortars.dm [hunk 2]
```diff
@@ -209,37 +211,44 @@
...
+				return
+			if(CEILING_IS_PROTECTED(A.ceiling, CEILING_PROTECTION_TIER_2) || protected_by_pylon(TURF_PROTECTION_MORTAR, T))
+				to_chat(user, SPAN_WARNING("You cannot hit the target. It is probably underground."))
+				return
+			if(SSticker.mode && MODE_HAS_TOGGLEABLE_FLAG(MODE_LZ_PROTECTION) && A.is_landing_zone)
+				to_chat(user, SPAN_WARNING("You cannot bomb the landing zone!"))
+				return
+
+		if(!ship_side)
+			var/turf/T1 = locate(T.x + pick(-1,0,0,1), T.y + pick(-1,0,0,1), T.z) //Small amount of spread so that consecutive mortar shells don't all land on the same tile
```
Comment (Zonespace27) [inline code/modules/cm_marines/equipment/mortar/mortars.dm:240]: same

File: code/modules/cm_marines/equipment/mortar/mortars.dm [hunk 3]
```diff
@@ -209,37 +211,44 @@
...
+			if(T1)
+				T = T1
+		else
+			var/crash_occurred = (SSticker?.mode?.is_in_endgame)
+			if(crash_occurred)
+				var/turf/our_turf = get_turf(src)
+				T = our_turf
+				travel_time = 0.5 SECONDS
+			else
+				to_chat(user, SPAN_RED("ARE YOU OUT OF YOUR MIND?!"))
```
Comment (Zonespace27) [inline code/modules/cm_marines/equipment/mortar/mortars.dm:250]: not really a great to_chat

File: code/modules/cm_marines/equipment/mortar/mortars.dm [hunk 4]
```diff
@@ -308,36 +317,52 @@
...
+			mortar_effect.icon = 'icons/obj/structures/mortar.dmi'
+			mortar_effect.icon_state = "mortar_ammo_custom"
+			mortar_effect.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+			mortar_effect.invisibility = INVISIBILITY_MAXIMUM
+			QDEL_IN(mortar_effect, 5 SECONDS)
+			notify_ghosts(header = "Custom Shell", message = "A custom mortar shell is about to land at [get_area(target)].", source = mortar_effect)
+
+		playsound(target, 'sound/weapons/gun_mortar_travel.ogg', 50, 1)
+		var/relative_dir
+		for(var/mob/M in range(15, target))
```
Comment (Zonespace27) [inline code/modules/cm_marines/equipment/mortar/mortars.dm:335]: no one letter vars (i know this isn't your code, but please fix it regardless)

File: code/modules/cm_marines/equipment/mortar/mortars.dm [hunk 5]
```diff
@@ -308,36 +317,52 @@
...
+			if(get_turf(M) == target)
+				relative_dir = 0
+			else
+				relative_dir = Get_Compass_Dir(M, target)
+			M.show_message( \
+				SPAN_DANGER("A SHELL IS COMING DOWN [SPAN_UNDERLINE(relative_dir ? uppertext(("TO YOUR " + dir2text(relative_dir))) : uppertext("right above you"))]!"), SHOW_MESSAGE_VISIBLE, \
+				SPAN_DANGER("YOU HEAR SOMETHING COMING DOWN [SPAN_UNDERLINE(relative_dir ? uppertext(("TO YOUR " + dir2text(relative_dir))) : uppertext("right above you"))]!"), SHOW_MESSAGE_AUDIBLE \
+			)
+		sleep(2.5 SECONDS) // Sleep a bit to give a message
+		for(var/mob/M in range(10, target))
```
Comment (Zonespace27) [inline code/modules/cm_marines/equipment/mortar/mortars.dm:345]: ditto

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/cm_marines/equipment/mortar/mortars.dm [hunk 6]
```diff
@@ -308,36 +317,52 @@
...
-	if(!protected_by_pylon(TURF_PROTECTION_MORTAR, target))
-		shell.detonate(target)
-	qdel(shell)
-	firing = FALSE
+	if(ship_side)
+		var/turf/our_turf = get_turf(src)
+		shell.detonate(our_turf)
+	else
+		if(istype(shell, /obj/item/mortar_shell/custom)) // big shell warning for ghosts
+			var/obj/effect/mortar_effect = new(target)
```
Comment (Zonespace27) [inline code/modules/cm_marines/equipment/mortar/mortars.dm:325]: i feel like this should be a subtype so you dont need to set lines 326-329 here

File: code/modules/cm_marines/equipment/mortar/mortars.dm (hunk 3 above)
Comment (kiVts) [reply code/modules/cm_marines/equipment/mortar/mortars.dm:250]: pretty sure its just a copypaste from c4 code

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**